bluemix/configuration/core_config/bx_config.go

```diff
@@ -296,11 +296,30 @@ func (c *bxConfig) IAMID() (guid string) {
 	return
 }
 
-func (c *bxConfig) IsLoggedIn() (loggedIn bool) {
-	c.read(func() {
-		loggedIn = c.data.IAMToken != ""
-	})
-	return
+// IsLoggedIn will check if the user is logged in. To determine if the user is logged both the access
+//
+// If token is near expiration or expired, and a refresh token is present attempt to refresh the token.
+// If token refresh was successful, check if the new iam token is valid. If valid, user is logged in,
+// otherwise user can be considered logged out. If refresh failed, then user is considered logged out.
+// If no refresh token is present, and token is expired, then user is considered logged out.
+func (c *bxConfig) IsLoggedIn() bool {
+	if token, refresh := c.IAMToken(), c.IAMRefreshToken(); token != "" || refresh != "" {
+		iamTokenInfo, refreshTokenInfo := NewIAMTokenInfo(token), NewIAMTokenInfo(refresh)
+		if iamTokenInfo.hasExpired() && refreshTokenInfo.exists() {
```

Comment:
IAM refresh tokens actually have a different structure than conventional IAM tokens, so the `exists()` method won't work here. I think we can just check if the refresh token is not empty.

Reply:
You're right. I will remove that check here